### PR TITLE
initial version of poc gc

### PR DESCRIPTION
**Pull request by evanmcc**
in order to have ledger fingerprints, we need to clean up old poc stuff that has never been written over.  This is a first, kind of ugly, attempt at a cleanup.

### CODE REVIEW

File: src/blockchain.erl [hunk 1]
```diff
@@ -552,7 +552,9 @@ add_block_(Block, Blockchain, Syncing) ->
                                                             lager:error("Error absorbing transaction, Ignoring Hash: ~p, Reason: ~p", [blockchain_block:hash_block(Block), Reason]),
                                                             Error;
                                                         ok ->
-                                                            case blockchain_ledger_v1:new_snapshot(blockchain:ledger(Blockchain)) of
+                                                            Ledger = blockchain:ledger(Blockchain),
+                                                            ok = blockchain_ledger_v1:maybe_gc_pocs(Ledger, Blockchain),
```
Comment (Vagabond) [inline src/blockchain.erl]: Wny not just pass the chain here and extract the ledger in the function?

Comment (evanmcc) [reply src/blockchain.erl]: it was originally written to just need the ledger, but then it needed the chain for some reason and I didn't clean it up very well.  will fix.

**Comment by Vagabond** [comment]
Can the draft status be removed?

File: src/ledger/v1/blockchain_ledger_v1.erl [hunk 2]
```diff
@@ -1150,6 +1151,65 @@ delete_pocs(OnionKeyHash, Ledger) ->
...
+                                fun(PoC) ->
+                                        H = blockchain_ledger_poc_v2:block_hash(PoC),
+                                        case H of
+                                            <<>> ->
+                                                %% pre-upgrade pocs are ancient
+                                                false;
+                                            _ ->
+                                                {ok, B} = blockchain:get_block(H, Chain),
+                                                BH = blockchain_block:height(B),
+                                                (Height - BH) < 100
```
Comment (vihu) [inline src/ledger/v1/blockchain_ledger_v1.erl]: Maybe the `100` should be configurable via some chain var? `poc_challenge_interval` perhaps?

File: src/blockchain_utils.erl [hunk 3]
```diff
@@ -26,6 +26,8 @@
 -include_lib("eunit/include/eunit.hrl").
 -endif.
 
+-include("blockchain_vars.hrl").
+
 -define(CHALLENGE_INTERVAL, poc_challenge_interval).
```
Comment (vihu) [inline src/blockchain_utils.erl]: Maybe rip this out if it's not getting used anymore?

**Pull request merged**